javaparser-core-generators/src/main/java/com/github/javaparser/generator/core/other/BndGenerator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
+ * Copyright (C) 2011, 2013-2016 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.generator.core.other;
+
+import com.github.javaparser.generator.Generator;
+import com.github.javaparser.utils.Log;
+import com.github.javaparser.utils.SourceRoot;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Generates the bnd.bnd file in javaparser-core.
+ */
+public class BndGenerator extends Generator {
+
+    private final Set<String> packages = new HashSet<>();
+
+    public BndGenerator(SourceRoot sourceRoot) {
+        super(sourceRoot);
+    }
+
+    @Override
+    public void generate() throws IOException {
+        Log.info("Running %s", () -> getClass().getSimpleName());
+        File root = sourceRoot.getRoot().toFile();
+        visit(root, "");
```

Comment:
SourceRoot was built for common file traversal tasks - since you're avoiding it, can you say what it is lacking?

Reply:
I didn't use it because it doesn't contain all the packages and files, probably because it only contains those files that are explicitly parsed by other generators. It doesn't offer a method to enumerate all the source files, including unparsed ones.

Reply:
Hmmm, maybe the generators should start by running SourceRoot over the whole project then... Thanks, that's for another time :-)